components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java

```diff
@@ -572,38 +575,69 @@ public void deactivateTenant(String tenantDomain) throws Exception {
      * @param tenantDomain The domain name of the tenant that needs to be deleted
      */
     public void deleteTenant(String tenantDomain) throws StratosException, org.wso2.carbon.user.api.UserStoreException {
+
         TenantManager tenantManager = TenantMgtServiceComponent.getTenantManager();
         if (tenantManager != null) {
             int tenantId = tenantManager.getTenantId(tenantDomain);
             try {
                 log.info("Starting Tenant Deletion process...");
 
                 notifyTenantDeletion(tenantId);
+                // Before deleting the tenant deactivate the tenant from the system.
+                // Deactivation also used to clear the tenant configuration context for that particular tenant.
+                deactivateTenant(tenantDomain);
                 TenantMgtUtil.deleteWorkernodesTenant(tenantId);
 
-                String tenantDelete = TenantMgtServiceComponent.getServerConfigurationService().getFirstProperty("TenantDelete");
+                String tenantDelete = TenantMgtServiceComponent.getServerConfigurationService()
+                        .getFirstProperty("TenantDelete");
 
                 if ((tenantDelete == null)
-                    && (tenantDelete.equals("true"))) {
+                    || (tenantDelete.equals("true"))) {
```

Comment:
Possible NPE, change the logic to if ((tenantDelete != null) && (tenantDelete.equals("true"))

Reply:
Or you can use just the "true".equals(tenantDelete) to check for null and String "true"